src/pages/patientView/clinicalInformation/lib/clinicalAttributesStyleConsts.js

```diff
@@ -0,0 +1,136 @@
+ * For instance, GENDER is displayed before OS_MONTHS.
+ * @param {string} a
+ * @param {string} b
+ */
+export const compare = (a, b) => {
+    return getOrder(a) - getOrder(b);
+};
+
+/**
+ * Builds a string to insert in an HTML element.
+ * @param {string} value
+ * @param {string} key
+ */
+export const stringBuilder = (value, key) => {
+    var builder = [];
+
+    const before = {
+        // Attributes with opening parenthesis
+        OS_MONTHS: '\xa0(',
+        DFS_MONTHS: '\xa0(',
+        CANCER_TYPE_DETAILED: '\xa0(',
+        PATIENT_DISPLAY_NAME: '\xa0(',
+        DERIVED_SAMPLE_LOCATION: '\xa0(',
+        SAMPLE_DISPLAY_NAME: '\xa0(',
+
+        // Text before an attribute
+        SERUM_PSA: ', Serum PSA: ',
+        ERG_FUSION_ACGH: ', ERG-fusion aCGH: ',
+        TMPRSS2_ERG_FUSION_STATUS: ', TMPRSS2-ERG Fusion: ',
+        GLEASON: ', Gleason: ',
+    };
+
+    const after = {
+        // Attributes with a closing parenthesis
+        CANCER_TYPE_DETAILED: ')',
+        PATIENT_DISPLAY_NAME: ')',
+        DERIVED_SAMPLE_LOCATION: ')',
+        SAMPLE_DISPLAY_NAME: ')',
+
+        // Text after an attribute
+        OS_MONTHS: ' months)',
+        DFS_MONTHS: ' months)',
+        AGE: ' years old',
+    };
+
+    // Show comma before clinical attributes, unless before non-breaking char and parenthesis
+    const prefix = before[key] === '\xa0(' ? '' : ', ';
+    builder.push(prefix);
+    builder.push(before[key]);
+    builder.push(value);
+    builder.push(after[key]);
+    return builder.join('');
+};
+
+/**
+ *  Gets a color for a given the key and value. 
+    If there a specified color for the key and value, return it.
+    If there is no such value, then get the color only specified for the key. 
+    Otherwise, return a default color. 
+ * @param {string} value 
+ * @param {string} key 
+ */
+export const colorGetter = (value, key) => {
+    // Style based on attr-id only and not attr-value
+    const color = 'black';
+    const NO_VALUE = '_';
+
+    const colors = {
+        sample_color_primary: 'black',
+        sample_color_recurrence: 'orange',
+        sample_color_metastasis: 'red',
+        sample_color_cfdna: 'blue',
+        sample_color_xenograft: 'pink',
+    };
+
+    // Attributes with special colors
+    const keys = {
+        OS_STATUS: {
+            DECEASED: '#f00',
+            DEAD: '#f00',
+            LIVING: 'rgb(0, 128, 0)',
+            ALIVE: 'rgb(0, 128, 0)',
+        },
+        DFS_STATUS: {
+            _: '#f00',
+            DiseaseFree: 'rgb(0, 128, 0)',
+            Yes: 'rgb(0, 128, 0)',
+        },
+        DERIVED_NORMALIZED_CASE_TYPE: {
+            Primary: colors.sample_color_primary,
+            Progressed: colors.sample_color_recurrence,
+            Recurrence: colors.sample_color_recurrence,
+            Metastasis: colors.sample_color_metastasis,
+        },
+    };
+
+    if (keys[key] != undefined) {
```

Comment:
you should always use !== instead of !=

Reply:
also, you could use ```key in keys```

Reply:
does !== also check for value and type?

Reply:
btw, we use !== for same reason as Luke cites above.  != does type coercion which we want to avoid.    in this case, it's not important, but just be in the habit of always using !== and ===.   safe to say, never use != and ==